File: app/src/main/java/q19/kenes/MainActivity.java
package q19.kenes;

import android.os.Bundle;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import q19.kenes_widget.KenesWidget;

class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button openWidget = findViewById(R.id.openWidget);

        openWidget();

        openWidget.setOnClickListener(v -> openWidget());
    }

    private void openWidget() {
        startActivity(KenesWidget.open(this, new KenesWidget.EntryParams(Configs.HOSTNAME)));
    }
}

File: app/src/main/java/q19/kenes/MainActivity.kt
package q19.kenes

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import q19.kenes_widget.KenesWidgetV2Activity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        openWidget()

        openWidget.setOnClickListener { openWidget() }
    }

    private fun openWidget() {
        startActivity(KenesWidgetV2Activity.newIntent(this, "https://kenes.vlx.kz"))
    }

}
